src/resolvers.ts

```diff
@@ -9,21 +23,27 @@ export default {
   },
 
   Mutation: {
-    createNote: (_: string, { sender, receiver, status, url }: INote) => {
-      //Need to pull users via slackID to add note
-      const note = new Note({ sender, receiver, status, url });
-      return note.save();
-      // have code pushing this note ID to sender - sent array
-      // reciever - recievedArray
-    },
-    createUser: (_: string, { slackID }: IUser) => {
-      const senderNotes: Array<string> = [];
-      const receiverNotes: Array<string> = [];
-      const user = new User({ slackID, senderNotes, receiverNotes });
-      return user.save();
-    },
-  },
-};
+    createNote: async(_:string, args:INote) => {
+      const {sender, receiver, status, url} = args
+  
+      const dbSender = await User.findOne({slackID: sender})
+      const dbReceiver = await User.findOne({slackID: receiver})
+
+      if (dbSender != null && dbReceiver != null) {
+        const note = await createNote(dbSender, dbReceiver, status, url)
+        if (note != undefined) {
+          dbSender.senderNotes.push(note._id)
+          dbReceiver.receiverNotes.push(note._id)
+          await dbSender.save()
+          await dbReceiver.save()
+        }
+        return note
+      }
```

Comment:
While this flow is readable now. I think it's too ineffecitive now. Can we be smarter in our use of awaits so that it doesn't block the flow unessessarily.